cosmrs/src/crypto/public_key.rs

```diff
@@ -101,22 +102,38 @@ impl TryFrom<&Any> for PublicKey {
     type Error = ErrorReport;
 
     fn try_from(any: &Any) -> Result<PublicKey> {
-        let tm_key = match any.type_url.as_str() {
+        match any.type_url.as_str() {
             Self::ED25519_TYPE_URL => {
-                let proto = proto::cosmos::crypto::ed25519::PubKey::decode(&*any.value)?;
-                tendermint::PublicKey::from_raw_ed25519(&proto.key)
+                proto::cosmos::crypto::ed25519::PubKey::decode(&*any.value)?.try_into()
             }
             Self::SECP256K1_TYPE_URL => {
-                let proto = proto::cosmos::crypto::secp256k1::PubKey::decode(&*any.value)?;
-                tendermint::PublicKey::from_raw_secp256k1(&proto.key)
+                proto::cosmos::crypto::secp256k1::PubKey::decode(&*any.value)?.try_into()
             }
             other => {
                 return Err(Error::Crypto)
```

Comment:
You can remove the `return` here (clippy will probably complain)

Reply:
Also I think it'd be good to preserve this:

```rust
Err(Error::Crypto).wrap_err_with(|| format!("invalid type URL for public key: {}", other)),
```

It ensures the error message contains the unrecognized type URL